FabNeso.py

```diff
@@ -0,0 +1,82 @@
+"""
+Task definitions for FabNESO plug-in to FabSIM software toolkit.
+
+Defines tasks for running simulations using Neptune Exploratory Software (NESO).
+"""
+
+from pathlib import Path
+
+try:
+    import fabsim.base.fab as fab
+except ImportError:
+    import base.fab as fab
+
+
+fab.add_local_paths("FabNeso")
+
+
+@fab.task
+@fab.load_plugin_env_vars("FabNeso")
+def neso(
+    config,
+    solver="Electrostatic2D3V",
+    conditions_file_name="two_stream_conditions.xml",
+    mesh_file_name="two_stream_mesh.xml",
+    wall_time_in="0:15:0",
+    memory_in="2G",
+    **args
+):
+    """
+    Run one NESO instance1
+
+    parameters:
+      - config: Directory with configuration information
+      - solver: Which NESO solver to use
+      - conditions_file_name: Path to conditions file
+      - mesh_file_name: Path to mesh file for NESO
+    """
+    fab.update_environment(args)
+    fab.with_config(config)
+    fab.execute(fab.put_configs, config)
+
+    fab.env.neso_solver = solver
+
+    # This we presumably change somehow so that it gets changed throughout
+    # the SWEEP dir?
+    fab.env.neso_conditions_file = (
+        Path(fab.find_config_file_path(config)) / conditions_file_name
+    )
+    # All of these should be in a config file somewhere
+    fab.env.neso_mesh_file = Path(fab.find_config_file_path(config)) / mesh_file_name
```

Comment:
After installing and trying this out locally I'm not sure these paths are set up as we'd need to. As far as I can tell the previous `fab.execute(fab.put_configs, config)` call copies the subdirectory of `config_files` named `config` to a directory on the machine being targetted (for now `localhost`). As this may be a remote machine I think the intention is that any paths we use in the templated script we run are from the locally copied `config` directory. This  sets `neso_mesh_file` and `neso_conditions_file` to point to the source configuration files rather than the local copy (and on a remote machine these paths won't exist).

Reply:
I've just taken out the fab.find_config_file_path() methods, and now I believe it has the desired behaviour. I'd misunderstood this method as finding the job directory after the configs had been transferred, rather than the original location of the configs.

Reply:
Yeah I had also initially assumed that was what `fab.find_config_file_path()` did.